imod/msw/model.py

```diff
@@ -242,3 +245,34 @@ def write(self, directory: Union[str, Path]):
         # write package contents
         for pkgname in self:
             self[pkgname].write(directory, index, svat)
+
+    def regrid_like(
+        self,
+        mf6_regridded_dis: mf6.StructuredDiscretization,
+        validate: bool = True,
+        regrid_context: Optional[RegridderWeightsCache] = None,
+        regridder_types: Optional[dict[str, Tuple[RegridderType, str]]] = None,
+    ) -> "MetaSwapModel":
+        unsat_database = self.simulation_settings["unsa_svat_path"]
+        regridded_model = MetaSwapModel(unsat_database)
+
+        target_grid = mf6_regridded_dis["idomain"]
+
+        mod2svat = self.pop("mod2svat", None)
+        grid = self.pop("grid", None)
```

Comment:
These keys are not hardcoded, but decided by the user. I think you have to find packages based on their type and pop them I fear...

Reply:
done